### PR TITLE
CoreList Block

### PR DESCRIPTION
**Pull request by blakewilson**
## Tasks

- [x] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.
- [x] If a code change, I have written testing instructions that the whole team & outside contributors can understand.
- [x] I have written and included a comprehensive [changeset](https://github.com/wpengine/faustjs/blob/canary/DEVELOPMENT.md#deployment) to properly document the changes I've made.

## Description

This PR implements the `CoreList` block in the blocks library.

## Testing

[You can reference this commit for the testing changes.](https://github.com/wpengine/faustjs/pull/1474/commits/cb5ee9cabcb2c1359bedf31458a6eb99809cd0d3)

**NOTE:** This PR relies on https://github.com/wpengine/wp-graphql-content-blocks/pull/123. Please ensure you are testing with this change in WPGraphQL Content Blocks before you begin testing this PR.

1. Generate the global stylessheet and import it into the project. `faust generateGlobalStylesheet`.
3. Copy your `theme.json` from your WordPress template (I'm using `twentytwentytwo`) into your Faust app.
4. Setup the Faust app with the theme.json and global stylesheet:
    ```_app.js
    import { WordPressBlocksProvider, fromThemeJson } from '@faustwp/blocks';
    import themeJson from '../theme.json';
    import '../globalStylesheet.css';
    
    export default function MyApp({ Component, pageProps }) {
      const router = useRouter();
      return (
        <FaustProvider pageProps={pageProps}>
          <WordPressBlocksProvider
            config={{
              blocks,
              theme: fromThemeJson(themeJson),
            }}>
            <Component {...pageProps} key={router.asPath} />
          </WordPressBlocksProvider>
        </FaustProvider>
      );
    }
    ```
5. Import the blocks in the `wp-blocks/index.js` file:
    ```js
    import { coreBlocks } from '@faustwp/blocks';
    
    export default {
      ...coreBlocks,
    };
    ```
6. Use the block fragment in the post query to get the `CoreList` data:
    ```graphql
    ${CoreBlocks.CoreList.fragments.entry}
    ...
    editorBlocks {
            __typename
            renderedHtml
            id: clientId
            parentClientId
            ...${CoreBlocks.CoreList.fragments.key}
          }
    ```
7. Use the `editorBlocks` from `props.data.post` in `WordPressBlocksViewer`:
    ```jsx
    <Container>
      <ContentWrapper>
        <WordPressBlocksViewer blocks={flatListToHierarchical(editorBlocks)}/>
      </ContentWrapper>
    </Container>
    ```
8. Go to WordPress site and create several quote blocks combining styles and colors.

### CODE REVIEW

**Review by matthewguywright — APPROVED** [review]

File: packages/blocks/tests/blocks/CoreList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+function renderProvider(props: CoreListFragmentProps) {
+  return render(
+    <WordPressBlocksProvider config={{ blocks: {}, theme: {} }}>
+      <CoreList {...props} />
+    </WordPressBlocksProvider>,
+  );
+}
+
+describe('<CoreCode />', () => {
```
Comment (theodesp) [inline packages/blocks/tests/blocks/CoreList.test.tsx:16]:
```suggestion
describe('<CoreList />', () => {
```

Comment (blakewilson) [reply packages/blocks/tests/blocks/CoreList.test.tsx]: Good catch!

**Comment by theodesp** [comment]
If you rebases the latest canary there is a fix for the broken audit-ci step

**Comment by blakewilson** [comment]
@theodesp Done!

**Pull request merged**